segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java

```diff
@@ -46,38 +47,34 @@ public IndexAppendProcessor(ScheduledExecutorService indexSegmentUpdateExecutor,
         this.store = store;
     }
 
-    /**
-     * Appends index segment on a separate thread.
-     * @param segmentName  segment name.
-     */
-    protected void processAppend(String segmentName) {
+    protected void processAppend(IndexAppend indexAppend) {
         // Not updating index segment for transient and transaction type.
-        if (isTransientSegment(segmentName) || isTransactionSegment(segmentName)) {
+        if (isTransientSegment(indexAppend.getSegment()) || isTransactionSegment(indexAppend.getSegment())) {
             return;
         }
-        LatestItemSequentialProcessor<String> latestItemProcessor = new LatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
-        latestItemProcessor.updateItem(segmentName);
+        LatestItemSequentialProcessor<IndexAppend> latestItemProcessor = new LatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
```

Comment:
This should be setup as a member variable in the constructor not re-created every time.